src/include/migraphx/raw_data.hpp

```diff
@@ -220,12 +220,12 @@ auto visit_all_pack(const shape& s, V1&& v1)
 /**
  * @brief Visits every object together
  * @details This will visit every object, but assumes each object is the same type. This can reduce
- * the deeply nested visit calls. This will return a function that will take the visitor callback.
- * So it will be called with `visit_all(xs...)([](auto... ys) {})` where `xs...` and `ys...` are the
+ * the deeply nested visit calls. Returns a function that takes the visitor callback.
+ * Calling syntax is `visit_all(xs...)([](auto... ys) {})` where `xs...` and `ys...` are the
  * same number of parameters.
  *
  * @param x A raw data object
- * @param xs Many raw data objects
+ * @param xs Many raw data objects.  All must be the same type as x.
```

Comment:
I think it confusing to say `All must be the same type as x` as they dont need to be the same C++ type, just the type used by the shape should be the same. I think it should just be removed.